pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/PartialUpsertHandler.java

```diff
@@ -90,4 +90,31 @@ public GenericRow merge(GenericRow previousRecord, GenericRow newRecord) {
     }
     return newRecord;
   }
+
+  protected PartialUpsertMerger getMergerForColumn(String column) {
+    return _column2Mergers.getOrDefault(column, _defaultPartialUpsertMerger);
+  }
+
+  public GenericRow merge(String column, Object previousValue, GenericRow newRecord) {
+    if (!_primaryKeyColumns.contains(column)) {
```

Comment:
These checks should be performed earlier

Reply:
thanks @Jackie-Jiang i have rewrite this interface, please take a look.